src/java.base/share/classes/jdk/internal/io/JdkConsoleImpl.java

```diff
@@ -348,19 +330,38 @@ public int read(char[] cbuf, int offset, int length)
         }
     }
 
-    ConsoleImpl() {
-        readLock = new Object();
-        writeLock = new Object();
-        out = StreamEncoder.forOutputStreamWriter(
-                new FileOutputStream(FileDescriptor.out),
-                writeLock,
-                CHARSET);
-        pw = new PrintWriter(out, true) { public void close() {} };
-        formatter = new Formatter(out);
-        reader = new LineReader(StreamDecoder.forInputStreamReader(
-                new FileInputStream(FileDescriptor.in),
-                readLock,
-                CHARSET));
-        rcb = new char[1024];
+    @Override
+    public JdkConsole console(boolean isTTY, Charset charset) {
+        if (isTTY) {
+            if (INSTANCE == null) {
```

Comment:
This would need a double checked locking, isn't it? Something like:
```
if (INSTANCE == null) {
   synchronized (JdkConsoleImpl.class) {
      if (INSTANCE != null) {
          return INSTANCE;
      }
      ...// rest of the current code that's in if block

```
Having said that perhaps the singleton instance isn't needed here and the `java.io.Console` deal with the caching?

Reply:
You are right. `JdkConsoleImpl` is instantiated within the static initializer of `Console`, there is no need to cache it. `Console.cons` is guaranteed to be a singleton. Removed tha caching altogther.